numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1540,62 +1524,34 @@ pack_inner(const char *inptr,
     npy_intp index = 0;
     int remain = n_in % 8;              /* uneven bits */
 
-#if defined NPY_HAVE_SSE2_INTRINSICS && defined HAVE__M_FROM_INT64
+#if NPY_SIMD
     if (in_stride == 1 && element_size == 1 && n_out > 2) {
-        __m128i zero = _mm_setzero_si128();
+        npyv_u8 v_zero = npyv_zero_u8();
         /* don't handle non-full 8-byte remainder */
         npy_intp vn_out = n_out - (remain ? 1 : 0);
-        vn_out -= (vn_out & 1);
-        for (index = 0; index < vn_out; index += 2) {
-            unsigned int r;
-            npy_uint64 a = *(npy_uint64*)inptr;
-            npy_uint64 b = *(npy_uint64*)(inptr + 8);
-            if (order == 'b') {
-                a = npy_bswap8(a);
-                b = npy_bswap8(b);
+        const int vstep = npyv_nlanes_u64;
+        vn_out -= (vn_out & (vstep - 1));
+        for (index = 0; index < vn_out; index += vstep) {
+            // Maximum paraller abillity: handle eight 64bits at one time
```

Comment:
```suggestion
            // Maximum parallel ability: handle eight 64-bit integers at one time
```
Perhaps worth adding `assert(vstep <= 8);` somewhere.

Reply:
`npyv_nlanes_u64` is generated from the inner macro,  which will not exceeds 8 for now.